src/common/mfu_io.c

```diff
@@ -253,6 +556,34 @@ int mfu_hardlink(const char* oldpath, const char* newpath)
 /*****************************
  * Files
  ****************************/
+void daos_open(const char* file, int flags,
+               int* mode_set, mode_t mode, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(file, &name, &dir_name);
+    assert(dir_name);
+
+    dfs_obj_t *parent = NULL;
+    int rc = dfs_lookup(mfu_file->dfs, dir_name, O_RDWR, &parent, NULL, NULL);
+    if (parent != NULL) {
+        rc = dfs_open(mfu_file->dfs, parent, name,
+                      S_IFREG | S_IWUSR | S_IRUSR, O_RDWR | O_CREAT,
```

Comment:
Can we use the mode settings passed from the user here, or is that not valid?

Reply:
@adammoody I updated this, I can use the mode passed in but also have to pass in a DAOS specific flag with it, S_IFREG | mode.

Reply:
I think adding S_IFREG should be fine.

We should also check whether we can use the ```flags``` parameter that is passed in from the user in place of the hardcoded ```O_RDWR | O_CREAT```.  However, we can do that in a future PR.

Reply:
@adammoody yeah, I already added the flags parameter, that one can be used.